android/src/main/java/nordpol/AndroidCard.java

```diff
@@ -22,17 +23,27 @@ private AndroidCard(IsoDep card) {
         this.card = card;
     }
 
+    /**
+     * Model names&numbers: 
+     * https://support.google.com/googleplay/android-developer/answer/6154891?hl=en 
+     */
+    private static boolean isSamsungS5() {
```

Comment:
@slomo I haven't completed testing yet, but I expect it's OK to always open&close. 

But for the sake of clean design and efficiency it's better if a device-specific workaround is not applied to all devices ;)